examples/base-next-app-router-tailwind-ts/src/app/globals.css

```diff
@@ -0,0 +1,28 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+@layer components {
+  /* This is an alternative to using function props and ownerState to style a component, the corresponding Switch component can be rendered in page.tsx without a use-client directive, but you cannot easily import the class names like in JS/TS.
+
+  However this may not be a viable pattern since you'll typically pass other non-serializable props like event handers to components, which means you will have to add your own 'use-client' boundaries in your app anyway. */
```

Comment:
@michaldudak do you think this is worth putting in the docs?

Reply:
I don't think so. I believe the way to use Base UI components with heavy customization (like using `slotProps`) will be done by wrapping the Base UI component in another custom component. I think it's reasonable to assume this wrapper will also have to be a client component.